File: Programers/Level2/기능개발.py
from collections import deque

def chkProcess(progresses, q):
    cnt = 0
    while q and progresses[q[0]] >= 100 :
        cnt +=1
        q.popleft()
        
    return cnt
    
def solution(progresses, speeds):
    answer = []
    q = deque()
    
    for i in range(len(progresses)):
        q.append(i)
    
    while q:
        for i in range(len(progresses)):
            progresses[i] += speeds[i]
            
        temp = chkProcess(progresses, q)
        if temp > 0:
            answer.append(temp)
                
    return answer

File: Javascript/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    const days = progresses.map((progress, index) =>
      Math.ceil((100 - progress) / speeds[index])
    );
    let cnt = 1;
    let maxDay = days[0];
  
    for (let i = 1; i < days.length; i++) {
      if (days[i] <= maxDay) {
        cnt++;
      } else {
        maxDay = days[i];
        answer.push(cnt);
        cnt = 1;
      }
    }
  
    answer.push(cnt);
  
    return answer;
  }